homeassistant/components/group/registry.py

```diff
@@ -60,11 +66,15 @@ def exclude_domain(self) -> None:
 
     def on_off_states(self, on_states: set, off_state: str) -> None:
         """Register on and off states for the current domain."""
+        domain = current_domain.get()
```

Comment:
In a follow-up PR, the context variable should be removed, and `domain` should be passed instead. There's a single custom integration published in HACS - `bodymiscale` - which also implements mixed group support so a breaking change + blog post + notifying the author of `bodymiscale` should be fine.